### PR TITLE
Adds HTML and CSV clipboard formats for CodeInspections copy

### PR DESCRIPTION
**Pull request by ThunderFrame**
CSV values are wrapped in double quotes if the value contains a comma or a double-quote
CSV values that contain double-quotes, have the double quotes escaped with a leading double-quote
CSV values that contain `\r\n`, have the `\r\n` replaced with `\n`

HTML tries to be XHTML compliant
HTML uses inline styles and nested DIVs in an effort to be compatible across email clients.
HTML values are HTML encoded using WebUtility.HtmlEncode
null values are encoded as NBSP.

HTML and CSV don't yet have column headers, so they're next TODO.

### CODE REVIEW

File: RetailCoder.VBE/Common/ExportFormatter.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Rubberduck.Common
+{
+    public static class ExportFormatter
+    {
```
Comment (retailcoder) [inline RetailCoder.VBE/Common/ExportFormatter.cs]: Could this be tweaked into an object that implements some `IExportFormatter` interface, constructor-injected as such wherever it's needed? The Test Explorer also uses `IClipboardWriter`...

File: RetailCoder.VBE/Inspections/ICodeInspectionResult.cs [hunk 2]
```diff
@@ -14,5 +14,6 @@ public interface ICodeInspectionResult : IComparable<ICodeInspectionResult>, ICo
         QualifiedSelection QualifiedSelection { get; }
         IInspection Inspection { get; }
         string ToCsvString();
```
Comment (retailcoder) [inline RetailCoder.VBE/Inspections/ICodeInspectionResult.cs]: This `ToCsvString` should be extracted into some `IExportable` interface and actually removed from `ICodeInspectionResult`, along with the new `ToArray()`.

File: RetailCoder.VBE/UI/CodeInspections/InspectionResultsViewModel.cs [hunk 3]
```diff
@@ -283,14 +285,24 @@ private void ExecuteCopyResultsCommand(object parameter)
...
                 ? RubberduckUI.CodeInspections_NumberOfIssuesFound_Singular
                 : RubberduckUI.CodeInspections_NumberOfIssuesFound_Plural;
-            var text = string.Format(resource, DateTime.Now, _results.Count) + Environment.NewLine + results;
 
-            _clipboard.Write(text);
+            var title = string.Format(resource, DateTime.Now.ToString(CultureInfo.InstalledUICulture), _results.Count);
+
+            var textResults = title + Environment.NewLine + string.Join("", _results.Select(result => result.ToString() + Environment.NewLine).ToArray());
+            var csvResults = ExportFormatter.Csv(aResults, title);
+            var htmlResults = ExportFormatter.HtmlClipboardFragment(aResults, title);
```
Comment (retailcoder) [inline RetailCoder.VBE/UI/CodeInspections/InspectionResultsViewModel.cs]: Ah, there's the coupling. If `InspectionResultsViewModel` took an `IExportFormatter` constructor parameter, its constructor would document the fact that the object is able to format/export its results.

**Comment by retailcoder** [comment]
Well done @ThunderFrame! I'm merging the PR, but there will be a few changes needed (see commit comments)

**Comment by ThunderFrame** [comment]
Yep, agreed on all of the above. I kept seeing possible refactors (and bad variable names), and there are more fixes I have in mind, but I was primarily focusing on getting the data into HTML. Cross-client HTML is *hard*.

I still need to add headers (including translations), and handle formatting for numbers and dates. I'm planning on adding RTF and Spreadsheet XML, as extra formats actually make clients less dependent on the HTML.

Hopefully @Hosch250 can do some WPF magic and make some export buttons.

**Comment by Hosch250** [comment]
@retailcoder does that window, but I can work on that after finish the SC window if he wants.

**Comment by ThunderFrame** [comment]
@Hosch250 I'll be adding the richer copy functionality to all windows, so if we add an export/save button, a common export dlaiog would be ideal.

**Pull request merged**